velox/functions/prestosql/aggregates/tests/MinMaxByAggregationTest.cpp

```diff
@@ -13,273 +13,988 @@
+      case TypeKind::BIGINT:                                         \
+        EXECUTE_TEST_BY_GLOBAL_OR_GROUP(                             \
+            testFunc, valueType, int64_t, isPartial);                \
+        break;                                                       \
+      case TypeKind::REAL:                                           \
+        EXECUTE_TEST_BY_GLOBAL_OR_GROUP(                             \
+            testFunc, valueType, float, isPartial);                  \
+        break;                                                       \
+      case TypeKind::DOUBLE:                                         \
+        EXECUTE_TEST_BY_GLOBAL_OR_GROUP(                             \
+            testFunc, valueType, double, isPartial);                 \
+        break;                                                       \
+      default:                                                       \
+        LOG(FATAL) << "Unsupported comparison type of minmax_by(): " \
+                   << mapTypeKindToName(GetParam().comparisonType);  \
+    }                                                                \
+  } while (0);
+
+#define EXECUTE_TEST(testFunc, isPartial)                         \
+  do {                                                            \
+    switch (GetParam().valueType) {                               \
+      case TypeKind::TINYINT:                                     \
+        EXECUTE_TEST_BY_VALUE_TYPE(testFunc, int8_t, isPartial);  \
+        break;                                                    \
+      case TypeKind::SMALLINT:                                    \
+        EXECUTE_TEST_BY_VALUE_TYPE(testFunc, int16_t, isPartial); \
+        break;                                                    \
+      case TypeKind::INTEGER:                                     \
+        EXECUTE_TEST_BY_VALUE_TYPE(testFunc, int32_t, isPartial); \
+        break;                                                    \
+      case TypeKind::BIGINT:                                      \
+        EXECUTE_TEST_BY_VALUE_TYPE(testFunc, int64_t, isPartial); \
+        break;                                                    \
+      case TypeKind::REAL:                                        \
+        EXECUTE_TEST_BY_VALUE_TYPE(testFunc, float, isPartial);   \
+        break;                                                    \
+      case TypeKind::DOUBLE:                                      \
+        EXECUTE_TEST_BY_VALUE_TYPE(testFunc, double, isPartial);  \
+        break;                                                    \
+      default:                                                    \
+        LOG(FATAL) << "Unsupported value type of minmax_by(): "   \
+                   << mapTypeKindToName(GetParam().valueType);    \
+    }                                                             \
+  } while (0);
+
+std::string rowVectorToString(const RowVectorPtr& rowVector) {
+  std::string str;
```

Comment:
Any particular reason not to use rowVector->toString(0, rowVector->size()) ?

Reply:
I am not aware of this utility. Thanks for pointing out!